### PR TITLE
return max insurable amount and deductible

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/java/com/hedvig/contract/web/InternalContractController.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/_/contracts")
+class InternalContractController(
+  private val contractService: ContractService
+) {
+  @GetMapping("/{memberId}/insurableAmountInfo")
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/web/InternalContractController.kt]: Maybe something like `/{memberId}/latest/insurableAmountInfo`

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt [hunk 2]
```diff
@@ -69,4 +73,23 @@ class ContractServiceImpl(
       RegenerateInsuranceCertificate(contractId = contract.id, agreementId = agreementId)
     )
   }
+
+  override fun getInsurableAmount(memberId: String, category: String): InsurableAmount {
```
Comment (johantjellden) [inline src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: This should probably be an enum.

File: src/main/java/com/hedvig/contract/web/InternalContractController.kt (hunk 1 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/web/InternalContractController.kt]: i would argue for` /i/members/{memberId}/contracts/latest/insurableAmountInfo`

File: src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt [hunk 3]
```diff
@@ -69,4 +73,23 @@ class ContractServiceImpl(
...
+          is ApartmentAgreementState -> if (agreement.lineOfBusiness.isStudent) 25_000 else 50_000
+          is HouseAgreementState -> 50_000
+          else -> throw Exception("what")
+        }
+      }
+    }
+
+    val deductible = Money.of(1500, "SEK")
+
+    return InsurableAmount(Money.of(maxAmount, "SEK"), deductible)
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: The currency should be based on the contract type i would say. So if ApartmentAgreementState then SEK if GreekApartmentInsurance then EUR

Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/service/ContractServiceImpl.kt]: I'll fix that for Norway etc.

File: src/main/java/com/hedvig/contract/web/InternalContractController.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/_/contracts")
+class InternalContractController(
+  private val contractService: ContractService
+) {
+  @GetMapping("/i/members/{memberId}/contracts/latest/insurableLimitsInfo")
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/web/InternalContractController.kt]:
```suggestion
  @GetMapping("/insurableLimitsInfo/member/{memberId}")
```

File: src/main/java/com/hedvig/contract/web/InternalContractController.kt [hunk 5]
```diff
@@ -15,7 +15,7 @@ import org.springframework.web.bind.annotation.RestController
 class InternalContractController(
   private val contractService: ContractService
 ) {
-  @GetMapping("/i/members/{memberId}/contracts/latest/insurableLimitsInfo")
+  @GetMapping("insurable/limits/member/{memberId}")
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/web/InternalContractController.kt]:
```suggestion
  @GetMapping("/members/{memberId}/insurableLimits")
```

**Pull request merged**